tensorflow/compiler/xla/service/cus_related_functions.cc

```diff
@@ -107,7 +55,7 @@ CUS_BINARY_OP(Div);
   llvm::Value* EmitCus##op(llvm::Value* lhs, llvm::Value* rhs,              \
                            llvm::IRBuilder<>* b) {                          \
     llvm::Module* module = b->GetInsertBlock()->getParent()->getParent();   \
-    llvm::StructType* cus = llvm_ir::getCusTy(module->getContext());        \
+    llvm::Type* cus = llvm_ir::getCusTy(module->getContext());        \
     llvm::Value* func =                                                     \
         module                                                              \
             ->getOrInsertFunction(                                          \
```

Comment:
shouldn't the return type of comparison operators be boolean rather than int8?

Reply:
I think there is no bool type in the llvm IR, it is usually represented as int8 (how clang compiles).

Reply:
int1 is usually what people use for representing boolean in llvm

Reply:
I actually used int1 at first, but it was causing problems when I registered symbols using simpleorcjit in TensorFlow. I just tested and int1 worked fine with the current way, will change it back

Reply:
I actually tried to use int1 at first, but int1 will cause problem in verifymodule, since when these bool functions are called in the kernel, the value that it assigns to expect an int8.

Reply:
You can keep it as int8 then.

I'm interested to know the error message from verifyModule with int1.

Reply:
I remember it was a store instruction, which means it expects the output to be int8 (but get an int1). I suspect that in TensorFlow, boolean size is 8 bits so it is addressable in the tensor, but I did not find any specification on boolean size of Tensorflow.

Reply:
PRED is the datatype for boolean in xla. And it is translated as int8 in the llvm. So I guess it is okay to use int8. May be LLVM backend for Nvidia GPUs don't have int1 support.